Source/msg.cpp

```diff
@@ -140,6 +140,7 @@ DLevel &GetDeltaLevel(uint8_t level)
 	auto emplaceRet = DeltaLevels.emplace(level, DLevel {});
 	assert(emplaceRet.second);
 	DLevel &deltaLevel = emplaceRet.first->second;
+	// The following memset should not be necessary, DLevel was constructed three lines up, the default constructor should set appropriate values.
 	memset(&deltaLevel, 0xFF, sizeof(DLevel));

```

Comment:
The reason for the `memset()` is so that the logic can compare `TCmdPItem::bCmd` with `CMD_INVALID` to see if a specific delta has been initialized. 🤮 

https://github.com/diasurgical/devilutionX/blob/2c7e886c8503f7e0c04085d8ff0900c168095684/Source/msg.cpp#L556-L560

Reply:
also uses monster.position.x being 255 as a test. Split the memsets for now.